### PR TITLE
std/lib/sort: update quicksort to use the new generics

### PR DESCRIPTION
**Pull request by pierrec**

### CODE REVIEW

File: lib/std/sort/quicksort.c3 [hunk 1]
```diff
@@ -1,99 +1,45 @@
...
-}
-
-fn void sort_ref_fn(Type list, ComparerRef cmp)
-{
-    usz len = sort::@len_from_list(list);
-    qsort_ref(list, 0, (isz)len - 1, cmp);
-}
-
-fn void sort(Type list) @if(ELEMENT_COMPARABLE)
+macro quicksort(list, cmp = null)
```
Comment (lerno) [inline lib/std/sort/quicksort.c3:4]: Does this really work when cmp is null?

**Comment by pierrec** [comment]
It does work with cmp set to null. Do you want a dedicated test for ti?

**Comment by lerno** [comment]
Thank you.

**Comment by lerno** [comment]
No, it's fine, I saw it working. I think quicksort could probably be a builtin since it's used so much by its name, so you can add a @builtin to its declaration if you want.

**Pull request merged**